### PR TITLE
DSS Monitor Complex Metrics

### PR DESCRIPTION
**Pull request by amarjandu**
Provide AWS `BucketBytesUploaded` / `BucketBytesDownloaded` to dss-monitor output.
Provide option for `--no-webhook` argument for json output to console

### CODE REVIEW

File: scripts/monitor_lambdas.py [hunk 1]
```diff
@@ -75,28 +105,57 @@ def send_slack_post(webhook:str, stages:dict):
...
-    secrets_store = os.environ['DSS_SECRETS_STORE']
-    if secret_name is None:
-        secret_name = 'monitor-webhook'
-    secret_id = f'{secrets_store}/{stage}/{secret_name}'
-    res = secretsmanager.get_secret_value(SecretId=secret_id)
-    return res['SecretString']
-
+def format_data_size(value: int):
+    base = 1000
+    bytes = float(value)
```
Comment (DailyDreaming) [inline scripts/monitor_lambdas.py]: `bytes` shadows a built-in: https://docs.python.org/3/library/functions.html#func-bytes .  Also `bytes` is typically an int so it should probably be renamed anyhow.

**Comment by amarjandu** [comment]
```
> ./scripts/monitor_lambda.py --no-webhook
{
    "dev": {
        "buckets": {
            "org-humancellatlas-dss-checkout-dev": {
                "BytesDownloaded": 0,
                "BytesUploaded": 0
            },
            "org-humancellatlas-dss-dev": {
                "BytesDownloaded": 0,
                "BytesUploaded": 0
            }
        },
        "lambdas": {
            "dss-admin-dev": {
                "Duration": 0,
                "Invocations": 0
            },
            "dss-checkout-sfn-dev": {
                "Duration": 0,
                "Invocations": 0
            },
            "dss-dev": {
                "Duration": 158395,
                "Invocations": 44889
            },
            "dss-dlq-reaper-dev": {
                "Duration": 57852557,
                "Invocations": 1440
            },
            "dss-gs-copy-sfn-dev": {
                "Duration": 34916,
                "Invocations": 126
            },
            "dss-gs-copy-write-metadata-sfn-dev": {
                "Duration": 10522,
                "Invocations": 33
            },
            "dss-index-dev": {
                "Duration": 147473,
                "Invocations": 7138
            },
            "dss-notify-dev": {
                "Duration": 327549460,
                "Invocations": 1441
            },
            "dss-notify-v2-dev": {
                "Duration": 48961,
                "Invocations": 7194
            },
            "dss-operations-dev": {
                "Duration": 0,
                "Invocations": 0
            },
            "dss-s3-copy-sfn-dev": {
                "Duration": 24861,
                "Invocations": 140
            },
            "dss-s3-copy-write-metadata-sfn-dev": {
                "Duration": 56786,
                "Invocations": 165
            },
            "dss-scalability-test-dev": {
                "Duration": 0,
                "Invocations": 0
            },
            "dss-sfn-launcher-dev": {
                "Duration": 16926,
                "Invocations": 166
            },
            "dss-sync-sfn-dev": {
                "Duration": 455151,
                "Invocations": 8127
            },
            "dss-visitation-dev": {
                "Duration": 0,
                "Invocations": 0
            }
        }
    }
}
> ./scripts/monitor_lambda.py

dev : 2019-06-10 10:09:45.908665 -> 2019-06-11 10:09:45.908665 |
 name | Invocations | Duration (seconds)

     | dss-admin-dev | 0 | 0.0 
     | dss-checkout-sfn-dev | 0 | 0.0 
     | dss-dev | 44888 | 158.408 
     | dss-dlq-reaper-dev | 1440 | 57852.994 
     | dss-gs-copy-sfn-dev | 126 | 34.916 
     | dss-gs-copy-write-metadata-sfn-dev | 33 | 10.522 
     | dss-index-dev | 7138 | 147.473 
     | dss-notify-dev | 1441 | 327547.287 
     | dss-notify-v2-dev | 7194 | 48.961 
     | dss-operations-dev | 0 | 0.0 
     | dss-s3-copy-sfn-dev | 140 | 24.861 
     | dss-s3-copy-write-metadata-sfn-dev | 165 | 56.786 
     | dss-scalability-test-dev | 0 | 0.0 
     | dss-sfn-launcher-dev | 166 | 16.926 
     | dss-sync-sfn-dev | 8127 | 455.151 
     | dss-visitation-dev | 0 | 0.0
Bucket | BytesUploaded | BytesDownloaded

     | org-humancellatlas-dss-dev | 0 Bytes | 0 Bytes
     | org-humancellatlas-dss-checkout-dev | 0 Bytes | 0 Bytes
```

File: scripts/monitor_lambdas.py [hunk 2]
```diff
@@ -75,28 +105,57 @@ def send_slack_post(webhook:str, stages:dict):
...
+            return (format + '%s') % ((base * bytes / unit), s)
+
+# conditionals
 if os.environ["DSS_DEPLOYMENT_STAGE"] is None:
     raise ValueError('Missing DSS_DEPLOYMENT_STAGE, exiting....')
     exit(1)
 
-stages = {f'{os.environ["DSS_DEPLOYMENT_STAGE"]}': None }
+# variables
+script_end_time = datetime.datetime.now()
```
Comment (DailyDreaming) [inline scripts/monitor_lambdas.py]: Is this really the `script_end_time`?

Comment (amarjandu) [reply scripts/monitor_lambdas.py]: ah i should rename that, for the AWS request, we would have to send the time period we want to look at, (starttime -> endtime)

Comment (amarjandu) [reply scripts/monitor_lambdas.py]: renamed

File: scripts/monitor_lambdas.py (hunk 1 above)
Comment (amarjandu) [reply scripts/monitor_lambdas.py]: renamed

**Comment by amarjandu** [comment]
This needs to be moved out of the DSS into another repo, and the architecture changed to subscribe to the DSS notification system, rather than iterate through logs.

However im not sure if the notification system can handle the bucket data input.... unless it  can be scraped from bundle meta-data.

**Comment by amarjandu** [comment]
migrated PR to 
https://github.com/HumanCellAtlas/data-store-monitor

**Pull request closed without merging**